### PR TITLE
Add @select

### PR DESCRIPTION
**Pull request by Snektron**
```zig
@select(
    comptime T: type,
    pred: std.meta.Vector(len, bool),
    a: std.meta.Vector(len, T),
    b: std.meta.Vector(len, T)
) std.meta.Vector(len, T)
```

Constructs a vector from a & b, based on the values in the predicate vector. For indices where the predicate value is true, the corresponding element from the a vector is selected, and otherwise from.

See also #903

### CODE REVIEW

**Comment by zigazeljko** [comment]
Why the name `@blend`? Both LLVM and WebAssembly call this operation `select`, since it just *selects* elements from the two vectors. `@blend` makes it sound like it interpolates (aka *blends*) between elements, like the [`mix`](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/mix.xhtml) function in GLSL.

**Comment by Snektron** [comment]
This is the name x86_64 uses. I picked blend initially because there is an existing proposal for async which would introduce a `select` keyword, and so didnt want to cause possible confusion between the two.

I dont have a strong opinion on the name though, im fine with renaming it to `@select`.

**Comment by nektro** [comment]
https://gobyexample.com/select
https://tour.golang.org/concurrency/5

https://llvm.org/docs/LangRef.html#select-instruction
https://webassembly.github.io/spec/core/syntax/instructions.html#syntax-instr-parametric

File: test/behavior/blend.zig [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const mem = std.mem;
+const expect = std.testing.expect;
+const Vector = std.meta.Vector;
+
+test "@shuffle" {
```
Comment (andrewrk) [inline test/behavior/blend.zig:7]: don't forget to rename this

**Review by andrewrk — CHANGES_REQUESTED** [review]
I'd like to request the name to be `@select` despite the fact that we have another accepted proposal for select syntax (#5263).

Other than that, this looks ready to merge. Nice work and thank you.

**Review by andrewrk — APPROVED** [review]

**Pull request merged**